### ISSUE TITLE
Updating with Basic Auth

### ISSUE THREAD

**Issue by Devligue** [Digital-Sapphire/PyUpdater#96]
Would it be possible to allow downloading updates from urls that require Basic Auth in the future? It shouldn't be too much work since it mostly require to add headers argument in:
https://github.com/JMSwag/PyUpdater/blob/3689dd3f82620eb34f5b8fe3b5ea0b2abcfb360a/pyupdater/client/downloader.py#L366-L368
so that it looks like this:
```
data = self.http_pool.urlopen('GET', file_url,
                              preload_content=False,
                              headers=self.headers,
                              retries=max_download_retries)
```

where:

```
self.headers = urllib3.util.make_headers(
    basic_auth='{}:{}'.format(UPDATE_URL_USER, UPDATE_URL_PASS))
```

of course user and password should be configurable and not hardcoded, it is just how I managed it to work for now 😉

**Comment by moscoquera** [comment]
this is supported since the last version (2.5.0)

**Comment by Devligue** [comment]
Ok, indeed it is mentioned as a feature and in change log, but otherwise I see no trace on how to implement this in the docs or frankly anywhere else. If I am blind, please insight me

**Comment by JMSwag** [comment]
@Devligue Totally my fault for not adding to the docs. I'll get this updated over the weekend. Today is my Friday 😄

**Comment by JMSwag** [comment]
Fixed in 0dcf3ba961a2e34af1f9949635627fefd5e4c967
